cpp/src/io/csv/datetime.cuh

```diff
@@ -322,11 +333,13 @@ __inline__ __device__ int64_t to_date_time(char const* begin, char const* end, b
   if (sep_pos != end) {
     if (extract_date(begin, sep_pos, dayfirst, &year, &month, &day)) {
       extract_time(sep_pos + 1, end, &hour, &minute, &second, &millisecond);
-      answer = seconds_since_epoch(year, month, day, hour, minute, second) * 1000 + millisecond;
+      answer = milliseconds_since_epoch(year, month, day, hour, minute, second) + millisecond;
     }
   } else {
     if (extract_date(begin, end, dayfirst, &year, &month, &day)) {
```

Comment:
`extract_date` should be refactored to return `cuda::std::chrono::year_month_day`.